### PR TITLE
Onboarding Activity, adapter, presenter implementation

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75248: Onboarding Activity, adapter, presenter implementation](https://edu-redmine.sun-asterisk.vn/issues/75248)
## WHAT
- Base Activity, Base Presenter
- Implement Onboarding's adapter, activity
- SharedPreferences for store user interests
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HN_Android_2024_Travenor_App/assets/165371062/3cf0b3f7-909c-4a2e-bdae-c44d60a12e77

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/travenor/screen/onboarding/OnboardingActivity.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    override fun onSaveUserSurveySuccess() {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
+        startActivity(intent)
+        finish()
+    }
+
+    override fun onSaveUserSurveyError(exception: Exception?) {
+        val msg = "Save user survey fail"
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/onboarding/OnboardingActivity.kt]: string

File: app/src/main/java/com/example/travenor/screen/onboarding/OnboardingActivity.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+    override fun userSurveyChangeListener(pair: Pair<String, Boolean>) {
+        val key = pair.first
+        val value = pair.second
+        mUserSurveyResult[key] = value
+    }
+
+    private fun saveUserSurveyResult() {
+        val placeList = mutableListOf<Place>()
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/onboarding/OnboardingActivity.kt:118]: mấy cái xử lí này có nên cho xuống presenter k nhỉ

Comment (daolq-2712) [reply app/src/main/java/com/example/travenor/screen/onboarding/OnboardingActivity.kt:118]: @thold-3445 
logic này vẫn là logic của view nên không move vào presenter e

File: app/src/main/java/com/example/travenor/screen/onboarding/adapter/OnboardingAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                OnboardingViewHolder(view)
+            }
+
+            USER_SURVEY_VIEW_TYPE -> {
+                val view = inflater.inflate(
+                    layout.onboarding_favorite_survey_layout,
+                    parent,
+                    false
+                )
+                return UserSurveyViewHolder(view)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/screen/onboarding/adapter/OnboardingAdapter.kt]: Dùng ViewBinding để code gọn em nhé!

File: app/src/main/java/com/example/travenor/screen/onboarding/UserInterestSurveyPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.example.travenor.screen.onboarding
+
+import com.example.travenor.constant.Food
+import com.example.travenor.constant.Place
+import com.example.travenor.data.sharedpreference.SharedPreferencesManager
+import java.io.IOException
+
+class UserInterestSurveyPresenter internal constructor(private val sharedPrefsManager: SharedPreferencesManager) :
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/screen/onboarding/UserInterestSurveyPresenter.kt]:
```suggestion
class UserInterestSurveyPresenter internal constructor(private val sharedPrefsManager: SharedPreferencesManager) :
```
Không pass trực tiếp ``SharedPreferencesManager`` vào Presenter mà thông qua ``Repository``
Trong trường hợp này là ``UserRepository``

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**